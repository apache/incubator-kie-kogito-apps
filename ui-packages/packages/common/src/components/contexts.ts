import React from 'react';

export const aboutLogoContext = React.createContext('');
