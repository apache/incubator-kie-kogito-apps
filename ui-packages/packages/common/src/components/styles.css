.kogito-common--aboutModalBox .pf-c-about-modal-box__hero {
    background-attachment: unset;
}

.kogito-common--PageLayout .pf-c-page__header-brand-link .pf-c-brand {
    --pf-c-page__header-brand-link--c-brand--MaxHeight: 42px;
}

.kogito-common__load-more {
    margin-bottom: var(--pf-global--spacer--lg);
    text-align: center;
}

.kogito-common--domain-explorer__table td:not(:first-of-type) {
    min-width: 10em;
}

.kogito-common__embedded-table  {
    --pf-c-table--nested--first-last-child--PaddingLeft: 0;
    margin-bottom: var(--pf-global--spacer--lg);
    max-width: 80vw;
}

.kogito-common--domain-explorer__table .pf-c-table__toggle{
    padding: 24px;
}

.kogito-common--domain-explorer__table-OverFlow {
    overflow-y: auto;
} 

.kogito-common--manage-columns__data-list .pf-c-data-list__toggle {
    margin-left: calc(-1 * var(--pf-global--spacer--md));
}

.kogito-common--manage-columns__data-list-content:not(:last-of-type) {
    box-shadow: none;   
}

.kogito-common--manage-columns__data-list .kogito-common--manage-columns__data-list-content .pf-c-data-list__expandable-content-body {
    padding: 0px;   
}

.kogito-common--manage-columns__data-list-content > .pf-c-data-list__expandable-content-body .pf-c-data-list__item-row {
    padding-left: calc( var(--pf-global--spacer--2xl) + var(--pf-global--spacer--lg));
}

.kogito-common--manage-columns__data-list-content .pf-c-data-list__cell {
    padding-left: var(--pf-global--spacer--md);
}

.kogito-common--manage-columns__modal .pf-c-dropdown {
    margin-bottom: var(--pf-global--spacer--md);
} 

.kogito-common--filter-options__inputs {
    margin-right: var(--pf-global--spacer--md);
}

.kogito-common--filter-options__inputs .pf-c-select__toggle-wrapper{
    min-width: 6em;
}