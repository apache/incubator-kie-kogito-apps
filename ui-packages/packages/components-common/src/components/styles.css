.kogito-components-common__load-more {
  margin-bottom: var(--pf-global--spacer--lg) !important;
  text-align: center;
}

.pf-c-spinner .pf-m-md .kogito-components-common__load-more-spinner {
  overflow: hidden;
  margin-left: 5px;
}

.kogito-components-common__form-footer-padding-top {
  padding-top: 20px;
}
