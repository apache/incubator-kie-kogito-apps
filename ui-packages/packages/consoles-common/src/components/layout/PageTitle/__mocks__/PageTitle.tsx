import React from 'react';

const MockedPageTitle = (): React.ReactElement => {
  return <></>;
};

export default MockedPageTitle;
