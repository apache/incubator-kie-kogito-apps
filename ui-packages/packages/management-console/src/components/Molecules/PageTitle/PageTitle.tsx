import React from 'react';
import { Title } from '@patternfly/react-core';
import { OUIAProps, componentOuiaProps } from '@kogito-apps/ouia-tools';

export interface IOwnProps {
  title: string;
}

const PageTitle: React.FC<IOwnProps & OUIAProps> = ({
  title,
  ouiaId,
  ouiaSafe
}) => {
  return (
    <React.Fragment>
      <Title
        headingLevel="h1"
        size="4xl"
        {...componentOuiaProps(ouiaId, 'page-title', ouiaSafe)}
      >
        {title}
      </Title>
    </React.Fragment>
  );
};

export default PageTitle;
