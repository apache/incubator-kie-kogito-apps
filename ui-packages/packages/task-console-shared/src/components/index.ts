export { default as TaskState } from './TaskState/TaskState';
