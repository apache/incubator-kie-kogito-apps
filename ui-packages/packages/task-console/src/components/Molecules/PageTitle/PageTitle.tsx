import React from 'react';
import { Flex, FlexItem, Title } from '@patternfly/react-core';
import { componentOuiaProps, OUIAProps } from '@kogito-apps/ouia-tools';

export interface IOwnProps {
  title: any;
  extra?: JSX.Element;
}

const PageTitle: React.FC<IOwnProps & OUIAProps> = ({
  title,
  extra,
  ouiaId,
  ouiaSafe
}) => {
  return (
    <React.Fragment>
      <Flex {...componentOuiaProps(ouiaId, 'page-title', ouiaSafe)}>
        <FlexItem spacer={{ default: 'spacerSm' }}>
          <Title headingLevel="h1" size="4xl">
            {title}
          </Title>
        </FlexItem>
        {extra ? (
          <FlexItem spacer={{ default: 'spacerSm' }}>{extra}</FlexItem>
        ) : null}
      </Flex>
    </React.Fragment>
  );
};

export default PageTitle;
