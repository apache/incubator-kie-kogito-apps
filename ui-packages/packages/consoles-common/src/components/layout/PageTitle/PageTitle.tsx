import React from 'react';
import { Flex, FlexItem } from '@patternfly/react-core/dist/js/layouts/Flex';
import { Title } from '@patternfly/react-core/dist/js/components/Title';
import {
  componentOuiaProps,
  OUIAProps
} from '@kogito-apps/ouia-tools/dist/utils/OuiaUtils';

export interface IOwnProps {
  title: any;
  extra?: JSX.Element;
}

export const PageTitle: React.FC<IOwnProps & OUIAProps> = ({
  title,
  extra,
  ouiaId,
  ouiaSafe
}) => {
  return (
    <Flex {...componentOuiaProps(ouiaId, 'page-title', ouiaSafe)}>
      <FlexItem spacer={{ default: 'spacerSm' }}>
        <Title headingLevel="h1" size="4xl">
          {title}
        </Title>
      </FlexItem>
      {extra ? (
        <FlexItem spacer={{ default: 'spacerSm' }}>{extra}</FlexItem>
      ) : null}
    </Flex>
  );
};
