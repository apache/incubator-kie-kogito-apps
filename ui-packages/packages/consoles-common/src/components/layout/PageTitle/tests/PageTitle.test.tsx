import React from 'react';
import { PageTitle } from '../PageTitle';
import { Label } from '@patternfly/react-core/dist/js/components/Label';
import { render, screen } from '@testing-library/react';

describe('PageTitle test', () => {
  it('default snapshot testing', () => {
    const { container } = render(<PageTitle title="Title" />);

    expect(container).toMatchSnapshot();
  });

  it('snapshot testing with extra', () => {
    const { container } = render(
      <PageTitle title="Title" extra={<Label>Label</Label>} />
    );

    expect(container).toMatchSnapshot();

    const extra = screen.getAllByText('Label');

    expect(extra).toBeTruthy();
  });
});
