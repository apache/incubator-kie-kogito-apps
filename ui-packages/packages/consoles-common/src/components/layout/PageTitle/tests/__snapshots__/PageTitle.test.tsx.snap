// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageTitle test default snapshot testing 1`] = `
<PageTitle
  title="Title"
>
  <Flex
    data-ouia-component-type="page-title"
    data-ouia-safe={true}
  >
    <div
      className="pf-l-flex"
      data-ouia-component-type="page-title"
      data-ouia-safe={true}
    >
      <FlexItem
        spacer={
          Object {
            "default": "spacerSm",
          }
        }
      >
        <div
          className="pf-m-spacer-sm"
        >
          <Title
            headingLevel="h1"
            size="4xl"
          >
            <h1
              className="pf-c-title pf-m-4xl"
              data-ouia-component-id="OUIA-Generated-Title-1"
              data-ouia-component-type="PF4/Title"
              data-ouia-safe={true}
            >
              Title
            </h1>
          </Title>
        </div>
      </FlexItem>
    </div>
  </Flex>
</PageTitle>
`;

exports[`PageTitle test snapshot testing with extra 1`] = `
<PageTitle
  extra={
    <Label>
      Label
    </Label>
  }
  title="Title"
>
  <Flex
    data-ouia-component-type="page-title"
    data-ouia-safe={true}
  >
    <div
      className="pf-l-flex"
      data-ouia-component-type="page-title"
      data-ouia-safe={true}
    >
      <FlexItem
        spacer={
          Object {
            "default": "spacerSm",
          }
        }
      >
        <div
          className="pf-m-spacer-sm"
        >
          <Title
            headingLevel="h1"
            size="4xl"
          >
            <h1
              className="pf-c-title pf-m-4xl"
              data-ouia-component-id="OUIA-Generated-Title-2"
              data-ouia-component-type="PF4/Title"
              data-ouia-safe={true}
            >
              Title
            </h1>
          </Title>
        </div>
      </FlexItem>
      <FlexItem
        spacer={
          Object {
            "default": "spacerSm",
          }
        }
      >
        <div
          className="pf-m-spacer-sm"
        >
          <Label>
            <span
              className="pf-c-label"
            >
              <span
                className="pf-c-label__content"
              >
                Label
              </span>
            </span>
          </Label>
        </div>
      </FlexItem>
    </div>
  </Flex>
</PageTitle>
`;
